src/compiler/checker.ts

```diff
@@ -14674,7 +14674,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         else {
             result.links.type = isUnion ? getUnionType(propTypes) : getIntersectionType(propTypes);
             if (writeTypes) {
-                result.links.writeType = isUnion ? getUnionType(writeTypes) : getIntersectionType(writeTypes);
+                result.links.writeType = isUnion ? getIntersectionType(writeTypes) : getUnionType(writeTypes);

```

Comment:
after [the original change](https://github.com/microsoft/TypeScript/pull/47674) has been merged it [has been mentioned in the comments](https://github.com/microsoft/TypeScript/pull/47674#discussion_r935892404) that likely this should work in the way I have adjusted it here but I don't see any open issue that would be the result of this discussion there

Reply:
1. Currently the getters and setter seem to be converted right away to plain readonly property (for getter) and plain property (for setter).  So will write types exist for setters at this point?
2. What about the case where `if (propTypes.length > 2){ ... }` before this `else` clause?  Isn't that branch  handled later in different code which also has to be updated?

Reply:
1. if I understand you correctly - what is displayed depends on the location. In write locations, the write type is displayed and in read locations, you get the read type. To change this the TS team would have to receive a well-motivated feature request with many examples of how it would differ from the current behavior.
2. Yes - you are right. I'll take a look at this.
3. What do you consider a plain type here? Could you rephrase this question?